core/trino-main/src/main/java/io/trino/security/AccessControlUtil.java

```diff
@@ -28,36 +28,35 @@
 {
     private AccessControlUtil() {}
 
-    public static void checkCanViewQueryOwnedBy(Identity identity, String queryOwner, AccessControl accessControl)
+    public static void checkCanViewQueryOwnedBy(Identity identity, Identity queryOwner, AccessControl accessControl)
     {
-        if (identity.getUser().equals(queryOwner)) {
+        if (identity.getUser().equals(queryOwner.getUser())) {
             return;
         }
         accessControl.checkCanViewQueryOwnedBy(identity, queryOwner);
     }
 
     public static List<BasicQueryInfo> filterQueries(Identity identity, List<BasicQueryInfo> queries, AccessControl accessControl)
     {
-        String currentUser = identity.getUser();
-        Set<String> owners = queries.stream()
+        Set<Identity> owners = queries.stream()
                 .map(BasicQueryInfo::getSession)
-                .map(SessionRepresentation::getUser)
-                .filter(owner -> !owner.equals(currentUser))
+                .map(SessionRepresentation::toIdentity)
+                .filter(owner -> !owner.getUser().equals(identity.getUser()))
                 .collect(toImmutableSet());
         owners = accessControl.filterQueriesOwnedBy(identity, owners);
 
-        Set<String> allowedOwners = ImmutableSet.<String>builder()
-                .add(currentUser)
+        Set<Identity> allowedOwners = ImmutableSet.<Identity>builder()
```

Comment:
This needs another object since `Identity` equality is just `user`, so we need to include roles and groups